beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -199,63 +194,49 @@ impl<E: EthSpec> PendingComponents<E> {
-                let expected_blobs_msg = block_kzg_commitments_count_opt
-                    .as_ref()
-                    .map(|num| num.to_string())
-                    .unwrap_or("unknown".to_string());
-
-                debug!(log,
-                    "Component(s) added to data availability checker";
-                    "block_root" => ?self.block_root,
-                    "received_block" => block_kzg_commitments_count_opt.is_some(),
-                    "received_blobs" => received_blobs,
-                    "expected_blobs" => expected_blobs_msg,
-                );
-
-                block_kzg_commitments_count_opt.map_or(false, |num_expected_blobs| {
-                    num_expected_blobs == received_blobs
-                })
+        // No data columns when there are 0 blobs
+        let expected_columns_opt = block_kzg_commitments_count_opt.map(|blob_count| {
+            if blob_count > 0 {
+                custody_column_count
+            } else {
+                0
             }
-            BlockImportRequirement::ColumnSampling(num_expected_columns) => {
-                // No data columns when there are 0 blobs
-                let expected_columns_opt = block_kzg_commitments_count_opt.map(|blob_count| {
-                    if blob_count > 0 {
-                        *num_expected_columns
-                    } else {
-                        0
-                    }
-                });
+        });
+        let expected_columns_msg = expected_columns_opt
+            .as_ref()
+            .map(|num| num.to_string())
+            .unwrap_or("unknown".to_string());
 
-                let expected_columns_msg = expected_columns_opt
-                    .as_ref()
-                    .map(|num| num.to_string())
-                    .unwrap_or("unknown".to_string());
+        let num_received_blobs = self.num_received_blobs();
+        let num_received_columns = self.num_received_data_columns();
 
-                let num_received_columns = self.num_received_data_columns();
+        debug!(
+            log,
+            "Component(s) added to data availability checker";
+            "block_root" => ?self.block_root,
+            "received_blobs" => num_received_blobs,
+            "expected_blobs" => expected_blobs_msg,
+            "received_columns" => num_received_columns,
+            "expected_columns" => expected_columns_msg,
+        );
 
-                debug!(log,
-                    "Component(s) added to data availability checker";
-                    "block_root" => ?self.block_root,
-                    "received_block" => block_kzg_commitments_count_opt.is_some(),
-                    "received_columns" => num_received_columns,
-                    "expected_columns" => expected_columns_msg,
-                );
+        let all_blobs_received = block_kzg_commitments_count_opt
+            .map_or(false, |num_expected_blobs| {
+                num_expected_blobs == num_received_blobs
```

Comment:
Is it worth switching to `>=` in case there's some future bug?
```suggestion
                num_received_blobs >= num_expected_blobs
```

Reply:
If we received more valid blobs than we should, then we probably want it to fail rather than covering it and importing them into the database?

Not sure how that's possible though, as blob sidecar fixed list is constructed here, and will fail if it does not have a corresponding kzg commitment:
https://github.com/sigp/lighthouse/blob/bf19769e0d7bf425c3b9850216ef10e7a611491e/beacon_node/beacon_chain/src/fetch_blobs.rs#L298-L304